### ISSUE TITLE
Coordinate transformations/arithmetic

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/react-msaview#27]
In order to realize the "dream" of this project, which is outlined in the roadmap, the code needs to be able to do some pretty weird coordinate arithmatic. The roadmap outlines going from SNPs, maybe on a genome browser track, to where that falls on a protein model. This would involve translating to protein space using genomic coordinate transformations/arithmetic

I stumbled on this paper which might be an interesting helper for going between these coordinate systems

https://www.biorxiv.org/content/10.1101/2021.07.09.451743v1.full.pdf

This is written in python and might need a js rewrite, or we just use in some other way just to test things out. Note that even this library is maybe not written to handle also translating into the MSA coordinate space which is really its own weird structure.

**Comment by ihh** [comment]
Yes there will need to be coordinate transformations (I've implemented this kind of thing too many times to count)

There are some intricacies but it is basically pretty straightforward - you need a map of (integer) nucleotide position to (integer) amino acid position, and another map from protein to PDB chain(s). And vice versa, and possibly a transitive through-map from genome to PDB that combines both of them. Mostly these could be arrays; there might be an argument for making the genome maps use deques (so insertions at the beginning are as cheap as insertions at the end). Other than that I don't think it should be too complex. ABrowse has the protein-sequence-to-PDB maps already.

**Comment by ihh** [comment]
I forgot one additional map, although you did mention it yourself: protein to MSA (and vice versa).

**Comment by ihh** [comment]
I'm not sure I would recommend trying to make this a re-usable library, since it's the sort of thing everyone ends up building for themselves. To put it another way, I've seen it reimplemented several times, often in a more heavyweight way than is strictly necessary, with the justification that this will facilitate re-use; but do people really re-use this?

Anyway I will butt out & let you get on with it ;)
